docs/RFCS/segmented_storage.md

```diff
@@ -0,0 +1,125 @@
+- Feature Name: segmented_storage
+- Status: draft
+- Start Date: 2015-07-29
+- RFC PR:
+- Cockroach Issue: #1644
+
+# Summary
+
+Partition the RocksDB keyspace into segments so that replicas created
+by raft replication do not share physical storage with replicas
+created by splits.
+
+# Motivation
+
+Currently, keys in the distributed sorted map correspond more or less
+directly to keys in RocksDB. This makes splits and merges cheap
+(since the bulk of the data does not need to be moved to a new
+location), but it introduces ambiguity since the same RocksDB key
+may be owned by different ranges at different times.
+
+For a concrete example of the problems this can cause (discussed more
+fully in
+[#1644](https://github.com/cockroachdb/cockroach/issues/1644)),
+consider a node `N3` which is temporarily down while a range `R1` is
+split (creating `R2`). When the range comes back up, the leaders of
+both `R1` and `R2` (`N1` and `N2` respectively) will try to bring it
+up to date. If `R2` acts first, it will see that `N3` doesn't have any
+knowledge of `R2` and so it sends a snapshot. The snapshot will replace
+data in `R2`'s keyspace, which `N3`'s replica of `R1` still covers.
+`N3` cannot correctly process any messages relating to `R2` until `R1`
+has caught up to the point of the split.
+
+# Detailed design
+
```

Comment:
It will be valuable to define the notion of Segment and what it is meant to represent. What follows in this discussion is the pure mechanism of how things work, and explaining the notion of a segment will make a valuable read.